crates/hyfetch/src/bin/hyfetch.rs

```diff
@@ -0,0 +1,1034 @@
+        };
+
+        Ok(Lightness::new(lightness)?)
+    }
+
+    let lightness = select_lightness().context("failed to select lightness")?;
+    debug!(?lightness, "selected lightness");
+    let color_profile = color_profile.with_lightness_adaptive(lightness, theme);
+    update_title(
+        &mut title,
+        &mut option_counter,
+        "Selected brightness",
+        &format!("{lightness:.2}", lightness = f32::from(lightness)),
+    );
+
+    //////////////////////////////
+    // 5. Color arrangement
+
+    let color_align: ColorAlignment;
+
+    // Calculate amount of row/column that can be displayed on screen
+    let (ascii_per_row, ascii_rows) = {
+        let (Width(term_w), Height(term_h)) =
+            terminal_size().context("failed to get terminal size")?;
+        let ascii_per_row = cmp::max(1, term_w / (u16::from(asc_width) + 2));
+        let ascii_per_row =
+            u8::try_from(ascii_per_row).expect("`ascii_per_row` should fit in `u8`");
+        let ascii_rows = cmp::max(1, (term_h - 8) / (u16::from(asc_lines) + 1));
+        let ascii_rows = u8::try_from(ascii_rows).expect("`ascii_rows` should fit in `u8`");
+        (ascii_per_row, ascii_rows)
+    };
+
+    // Displays horizontal and vertical arrangements in the first iteration, but
+    // hide them in later iterations
+    let hv_arrangements = [
+        ("Horizontal", ColorAlignment::Horizontal { fore_back }),
+        ("Vertical", ColorAlignment::Vertical { fore_back }),
+    ];
+    let mut arrangements: IndexMap<Cow<str>, ColorAlignment> =
+        hv_arrangements.map(|(k, ca)| (k.into(), ca)).into();
+
+    // Loop for random rolling
+    let mut rng = fastrand::Rng::new();
+    loop {
+        clear_screen(Some(&title), color_mode, debug_mode).context("failed to clear screen")?;
+
+        // Random color schemes
+        let mut preset_indices: Vec<PresetIndexedColor> =
+            (0..color_profile.unique_colors().colors.len())
+                .map(|pi| u8::try_from(pi).expect("`pi` should fit in `u8`").into())
+                .collect();
+        let slots: IndexSet<NeofetchAsciiIndexedColor> = {
+            let ac = NEOFETCH_COLORS_AC
+                .get_or_init(|| AhoCorasick::new(NEOFETCH_COLOR_PATTERNS).unwrap());
+            ac.find_iter(&asc)
+                .map(|m| {
+                    asc[m.start() + 3..m.end() - 1]
+                        .parse()
+                        .expect("neofetch ascii color index should not be invalid")
+                })
+                .collect()
+        };
+        while preset_indices.len() < slots.len() {
+            preset_indices.extend_from_within(0..);
+        }
+        let preset_index_permutations: IndexSet<Vec<PresetIndexedColor>> = preset_indices
+            .into_iter()
+            .permutations(slots.len())
+            .collect();
+        let random_count = cmp::max(
+            0,
+            usize::from(ascii_per_row) * usize::from(ascii_rows) - arrangements.len(),
```

Comment:
When the terminal width can only fit one ascii art, this will underflow (unsigned `1 - 2`) , and then later panic on the u8 conversion.